app/views/entities/new.html.erb

```diff
@@ -0,0 +1,54 @@
+<div class='container'>
+  <%= form_with model: @entity,  url: user_group_entities_path(params[:user_id], params[:group_id]), method: 'post', class: "flex-column form", id:'transaction_form' do |form| %>
+    <header class="header flex-row">
+      <div>
+        <%= link_to user_group_path(params[:user_id],  params[:group_id]) do %>
+            <%= image_tag('left-arrow.png', size: "20x20") %>
+          <% end %>
+      
+      </div>
+      <div class="title">
+        NEW TRANSACTION
+      </div>
+      <div>
+        <button id="submit_button" class='submit' >Save</button>
+      </div>
+
+    </header>
+    <%= render partial: 'shared/messages' %>
+    <%= form.text_field :name, id:'name', autofocus: true, placeholder:'Transaction name', required: true %>
+    <%= form.number_field :amount, id:'amount', step: 0.01, min:0.01, placeholder:'Amount', required: true %>
+    <%= form.hidden_field :checked, id:'checked-items' %>
+    
+    <%= field_set_tag 'Categories', class:'checkboxes' do %>
+      <%= form.collection_check_boxes :group_ids, @groups, :id, :name, { hide_label: true } %>
+    <% end %>
+    <% end %>
+</div>
+
+<script>
+  let checked_items = document.querySelector('#checked-items')
+  let name = document.querySelector('#name')
+  let amount = document.querySelector('#amount')
+  let submit_button = document.querySelector('#submit_button')
+
+  setInterval(() => {
+    let checkboxes = document.querySelectorAll('input[type="checkbox"]:checked')
+    checked_items.value = checkboxes.length
+  }, 100);
+
+
+
+const form = document.getElementById('transaction_form');
+form.addEventListener('submit', (event) => { 
+  event.preventDefault()
+  });
+
+function validateSubmit() {
+  if (checked_items.value > 0 && name.value.trim().length>0 && amount.value > 0) {
+    form.submit()
+  }
+}
```

Comment:
- [ ] When two categories checkboxes are checked and the transaction is submitted, it takes both categories and it adds the transaction to both the categories. I think this is not the correct behavior of the app. Kindly try to handle it in `validateSubmit` function or maybe somewhere else. :pray: 

![image](https://user-images.githubusercontent.com/64894927/144649124-d2e5d852-eded-45e8-9419-92132e7c9fd1.png)

![image](https://user-images.githubusercontent.com/64894927/144649190-3eca90e0-c5f2-4c47-9747-cf3c62195632.png)

![image](https://user-images.githubusercontent.com/64894927/144649262-26d34375-e050-4a55-aba3-82fcd0f35d97.png)

Reply:
Hi @henatan99,
Thank you for your time and effort on the review.
Logically, this is not the correct behavior, but according to the terms of the project, it should be so. 
According to the ERD,  Entity and Group models have many-to-many relations, and according to project requirements for "Add a new transaction" page: 

The user fills out a form to create a new transaction with:
- name (mandatory)
- amount (mandatory)
- **categories (mandatory at least one)** .
So users can select more than one category. 

Should I change this behavior?

Reply:
Hi @toliboff , I think it has nothing to do with the ERD, models, or associations. You can handle this issue on the form, maybe radio buttons can be used or any other means which returns a warning when more than one checkboxes are selected during form submission. :pray: